src/core/src/op/fake_convert.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/fake_convert.hpp"
+
+#include "itt.hpp"
+
+namespace ov {
+namespace op {
+namespace v13 {
+namespace fake_convert {
+static const std::vector<std::string>& get_valid_types() {
+    static const std::vector<std::string> valid_types{"f8e4m3", "f8e5m2"};
+    return valid_types;
+}
+}  // namespace fake_convert
+
+FakeConvert::FakeConvert(const ov::Output<ov::Node>& arg,
+                         const ov::Output<ov::Node>& scale,
+                         std::string destination_type)
+    : Op({arg, scale}),
+      m_destination_type(std::move(destination_type)) {
+    constructor_validate_and_infer_types();
+}
+
+FakeConvert::FakeConvert(const ov::Output<ov::Node>& arg,
+                         const ov::Output<ov::Node>& scale,
+                         const ov::Output<ov::Node>& shift,
+                         std::string destination_type)
+    : Op({arg, scale, shift}),
+      m_destination_type(std::move(destination_type)) {
+    constructor_validate_and_infer_types();
+}
+
+const std::string& FakeConvert::get_destination_type() const {
+    return m_destination_type;
+}
+
+void FakeConvert::validate_and_infer_types() {
+    OV_OP_SCOPE(v13_FakeConvert_validate_and_infer_types);
+    validate_type();
+    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+}
+
+std::shared_ptr<ov::Node> FakeConvert::clone_with_new_inputs(const ov::OutputVector& new_args) const {
+    OV_OP_SCOPE(v13_FakeConvert_clone_with_new_inputs);
+    if (new_args.size() == 2) {
+        return std::make_shared<FakeConvert>(new_args.at(0), new_args.at(1), m_destination_type);
+    } else if (new_args.size() == 3) {
+        return std::make_shared<FakeConvert>(new_args.at(0), new_args.at(1), new_args.at(2), m_destination_type);
```

Comment:
@mitruska, let me put it here but it is not a blocker for this PR, just a general comment for Op enabling that can be made simpler: `clone_with_new_inputs` can be generated automatically with using default ctor that is available for all operations, and the visitors. It is applicable for all the operations and utilizing such a mechanism will simplify the process of new ops enabling. I've triggered on this condition that just a consequence of having two ctors with a different number of arguments. Various ctors are provided for externals op usage for convenience in C++, but for such a standard procedure as cloning, it makes sense to use more low-level approach. Compare with serialization/deserialization: they are implemented automatically using the same low-level apparatus.